mlrun/common/schemas/alert.py

```diff
@@ -214,3 +214,77 @@ def is_valid(cls, value: str) -> bool:
     @classmethod
     def valid_intervals(cls) -> list:
         return list(cls._value2member_map_.keys())
+
+    @classmethod
+    def from_function(cls, partition_function: str):
+        """
+        Returns the corresponding PartitionInterval for a given partition function,
+        or None if the function is not mapped.
+
+        :param partition_function: The partition function to map to an interval.
+        :return: PartitionInterval corresponding to the function, or None if no match is found.
+        """
+        partition_function_to_partitions_interval = {
+            "DAY": "DAY",
+            "DAYOFMONTH": "DAY",
+            "MONTH": "MONTH",
+            "YEARWEEK": "YEARWEEK",
+        }
+        interval = partition_function_to_partitions_interval.get(partition_function)
+        if interval and cls.is_valid(interval):
+            return cls[interval]
+        return None
+
+    def get_partition_info(
+        self,
+        partition_datetime: datetime,
+    ) -> tuple[str, str, str]:
+        """
+        Generates partition details for a specified datetime.
+        :param partition_datetime: The datetime used for generating partition details.
+
+        :return: A tuple containing:
+            - partition_name: The name for the partition.
+            - partition_value: The "LESS THAN" value for the next partition boundary.
+            - partition_expression: The SQL partition expression.
+        """
+
+        partition_name = self.get_partition_name(partition_datetime)
+        partition_boundary_date = self.get_next_partition_time(partition_datetime)
+        partition_value = self.get_partition_name(partition_boundary_date)
+        partition_expression = self.get_partition_expression()
+
+        return partition_name, partition_value, partition_expression
+
+    def get_next_partition_time(self, current_datetime: datetime) -> datetime:
```

Comment:
add a comment to explain this method's purpose and how the next partition time is calculated as it is confusing by only reading the implementation.